File: problem-125-easy.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        if (s.size() == 0)
            return true;
            
        int left = 0;
        int right = s.size() - 1;
        
        while (left <= right) {
            if (!isalpha(s[left]) && !isdigit(s[left]))
                left++;
            else if (!isalpha(s[right]) && !isdigit(s[right]))
                right--;
            else {
                if (tolower(s[left]) == tolower(s[right])) {
                    left++;
                    right--;
                }
                else return false;
            }
        }

        return true;
    }
};

File: problem-125-easy.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if len(s) == 0:
            return True
        
        left = 0
        right = len(s) - 1

        while (left <= right):
            if (not s[left].isalnum()):
                left += 1
            elif (not s[right].isalnum()):
                right -= 1
            else:
                if (s[left].lower() == s[right].lower()):
                    left += 1
                    right -= 1
                else:
                    return False
    
        return True
    
        
